engine/src/test/scala/cromwell/jobstore/JobStoreServiceSpec.scala

```diff
@@ -29,38 +30,35 @@ class JobStoreServiceSpec extends CromwellTestKitWordSpec with Matchers with Moc
       val jobStoreService = system.actorOf(JobStoreActor.props(jobStore))
 
       val workflowId = WorkflowId.randomId()
-      val successCall = mock[WdlTaskCall]
-      successCall.fullyQualifiedName returns "foo.bar"
-      val mockTask = mock[WdlTask]
-      mockTask.outputs returns Seq(TaskOutput("baz", WdlStringType, EmptyExpression, mock[Ast], Option(mockTask)))
-      successCall.task returns mockTask
+      val mockTask = WomMocks.mockTaskDefinition("bar")
+      .copy(outputs = Set(OutputDefinition("baz", WdlStringType, EmptyExpression))) 
+      val successCall = WomMocks.mockTaskCall("bar", definition = mockTask)
 
       val successKey = BackendJobDescriptorKey(successCall, None, 1).toJobStoreKey(workflowId)
 
-      jobStoreService ! QueryJobCompletion(successKey, mockTask.outputs)
+      jobStoreService ! QueryJobCompletion(successKey, mockTask.outputs.toSeq)
       expectMsgType[JobNotComplete.type](MaxWait)
 
       val outputs = Map("baz" -> JobOutput(WdlString("qux")))
 
       jobStoreService ! RegisterJobCompleted(successKey, JobResultSuccess(Option(0), outputs))
       expectMsgType[JobStoreWriteSuccess](MaxWait)
 
-      jobStoreService ! QueryJobCompletion(successKey, mockTask.outputs)
+      jobStoreService ! QueryJobCompletion(successKey, mockTask.outputs.toSeq)
       expectMsgPF(MaxWait) {
         case JobComplete(JobResultSuccess(Some(0), os)) if os == outputs =>
       }
 
-      val failureCall = mock[WdlTaskCall]
-      failureCall.fullyQualifiedName returns "baz.qux"
+      val failureCall = WomMocks.mockTaskCall("qux")
       val failureKey = BackendJobDescriptorKey(failureCall, None, 1).toJobStoreKey(workflowId)
 
-      jobStoreService ! QueryJobCompletion(failureKey, mockTask.outputs)
+      jobStoreService ! QueryJobCompletion(failureKey, mockTask.outputs.toSeq)
       expectMsgType[JobNotComplete.type](MaxWait)
 
       jobStoreService ! RegisterJobCompleted(failureKey, JobResultFailure(Option(11), new IllegalArgumentException("Insufficient funds"), retryable = false))
       expectMsgType[JobStoreWriteSuccess](MaxWait)
 
-      jobStoreService ! QueryJobCompletion(failureKey, mockTask.outputs)
+      jobStoreService ! QueryJobCompletion(failureKey, mockTask.outputs.toSeq)
```

Comment:
Perhaps more appropriate to change `QueryJobCompletion` to take a `Set`?

Reply:
I think we want the outputs to be a List actually, same way the inputs are, so they can be evaluated in order. When they become a List in WOM the toSeq will be unnecessary

Reply:
Oh, these are task outputs. Yeah I agree!